server/index.js

```diff
@@ -0,0 +1,129 @@
+require('dotenv').config();
+const express = require('express');
+const helmet = require('helmet');
+const cors = require('cors');
+const server = express();
+const jwt = require('jsonwebtoken');
+
+const bcrypt = require('bcryptjs');
+server.use(helmet());
+server.use(cors());
+server.use(express.json());
+const db = require('./data/dbConfig.js'); // database configuration file location
+
+// CONFIG: server settings
+const serverPort = process.env.PORT || 7100; // server port
+const serverName = `auth-i`; // Name of server to display at "/" endpoint 
+const projectPullRequest = `https://github.com/LambdaSchool/auth-ii/pull/215`;
+
+// CONFIG: endpoint routing
+// const users = require('./data/routes/usersRoutes');
+// server.use('/api/users', users);
+
+// CONFIG: Models
+const users = require('./data/models/usersModels');
+
+// ENDPOINTS
+server.get('/', (req, res) => { // sanity check root endpoint
+  res.send(`${serverName} running on port ${serverPort}<br>More information: <a href="${projectPullRequest}">GitHub Repo</a>`);
+});
+
+const jwtSecret = process.env.JWT_SECRET || 'add a secret to your .env file with this key';
+
+function generateToken(user) { // Generates token for cookie
+
+  const jwtPayload = { // payload sets the information that will be sent through the cookie (NOTE SECURE)
+    ...user,
+    hello: `${user}`,
+    role: 'admin'
+  };
+
+  jwtOptions = {
+    expiresIn: '1m'
+  };
+
+  return jwt.sign(jwtPayload, jwtSecret, jwtOptions);
+}
+
+server.post('/api/login', (req, res) => { // api login endpoint
+  const creds = req.body; // store body of post request in credentials variable
+
+  db('users').where({ username: creds.username }) // search users db for username
+    .first() // return first result
+    .then(user => {
+      if (user && bcrypt.compareSync(creds.password, user.password)) {
+        const token = generateToken(user); // LOOK BACK AT THIS (might change to username)
+        // check if user exists and user bcrypt hashed password with submitted password
+        res.status(200).json({ message: `Authentication success. Welcome ${user.username}.`, token })
+      } else {
+        res.status(401).json({ message: 'Authentication failed.' })
+      }
+    })
+    .catch(err => res.status(500).json({ err }));
+});
+
+server.post('/api/register', (req, res) => { // api register endpoint
+  const credentials = req.body; // store body of post request in credentials variable
+  const hash = bcrypt.hashSync(credentials.password, 14) // hash the password
+  credentials.password = hash; // store hashed password on the credentials object
+  db('users')
+    .insert(credentials)
+    .then(ids => {
+      const id = ids[0];
+      res.status(201).json({ newUserId: id });
+    })
+    .catch(err => {
+      res.status(500).json(err);
```

Comment:
Write an error massage that says why the user could not register and how to fix the problem.